src/common/src/catalog/column.rs

```diff
@@ -291,6 +297,25 @@ impl ColumnCatalog {
     }
 }
 
+pub fn columns_extend(preserved_columns: &mut Vec<ColumnCatalog>, columns: Vec<ColumnCatalog>) {
```

Comment:
Does `ColumnIdGenerator` help here? See `check_and_add_timestamp_column`.

Reply:
> Does `ColumnIdGenerator` help here? See `check_and_add_timestamp_column`.

We wish proto columns would remain their own ids invariant